src/block_reporting.js

```diff
@@ -0,0 +1,21 @@
+import * as Blockly from 'blockly/core';
+import {Colours} from '../core/colours.js';
+
+export function reportValue(id, value) {
+  const block = Blockly.getMainWorkspace().getBlockById(id) ||
+      Blockly.getMainWorkspace().getFlyout().getWorkspace().getBlockById(id);
+  const field = block.inputList[0].fieldRow[0];
+  if (!block) {
+    throw 'Tried to report value on block that does not exist.';
+  }
```

Comment:
I think you need to swap these statements. Otherwise we're going to get an error from dereferencing `blockly.inputList` before we get to your fancy error.

Reply:
Done